A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js

```diff
@@ -1,51 +1,99 @@
         console.log('A32NX_GPWS init');
 
-        SimVar.SetSimVarValue("L:GPWS_SINK_RATE", "Bool", 0);
-        SimVar.SetSimVarValue("L:GPWS_PULL_UP", "Bool", 0);
-        SimVar.SetSimVarValue("L:GPWS_DONT_SINK", "Bool", 0);
-        SimVar.SetSimVarValue("L:GPWS_TOO_LOW", "Enum", 0); // 0 (Off) 1 (Too Low Terrain) 2 (Too Low Gear) 3 (Too Low Flaps)
+        this.radnav.init(NavMode.FOUR_SLOTS);
 
-        //SimVar.SetSimVarValue("L:GPWS_MODE_2", "Enum", 0); // GPWS Mode 2 = 0 (Off) 1 (Terrain Terrain Pull Up) 2 (Terrain)
-        //SimVar.SetSimVarValue("L:GPWS_MODE_5", "Enum", 0); // GPWS Mode 5 = 0 (Off)
+        SimVar.SetSimVarValue("L:A32NX_GPWS_GS_Warning_Active", "Bool", 0);
+        SimVar.SetSimVarValue("L:A32NX_GPWS_Warning_Active", "Bool", 0);
     }
 
     update(deltaTime, _core) {
         this.faults();
-        this.gpws();
+        this.gpws(deltaTime);
     }
+    gpws(deltaTime) {
+        const radioAlt = SimVar.GetSimVarValue("PLANE ALT ABOVE GROUND MINUS CG", "Feet");
+
+        this.UpdateAltState(radioAlt);
+        this.differentiate_radioalt(radioAlt, deltaTime);
 
-    gpws() {
-        const SYS_PushButton = SimVar.GetSimVarValue("L:PUSH_OVHD_GPWS_SYS", "Bool");
-        const FLAP_PushButton = SimVar.GetSimVarValue("L:PUSH_OVHD_GPWS_FLAP", "Bool");
-        const altitude = Simplane.getAltitudeAboveGround();
-        const vSpeed = Simplane.getVerticalSpeed();
-        const Airspeed = SimVar.GetSimVarValue("AIRSPEED INDICATED", "Knots");
         const mda = SimVar.GetSimVarValue("L:AIRLINER_MINIMUM_DESCENT_ALTITUDE", "Number");
         const dh = SimVar.GetSimVarValue("L:AIRLINER_DECISION_HEIGHT", "Number");
         const phase = SimVar.GetSimVarValue("L:AIRLINER_FLIGHT_PHASE", "Enum");
 
-        if (!SYS_PushButton && altitude >= 10 && altitude <= 2450) { //Activate between 10 - 2450 radio alt unless SYS is off
-            this.gpws_pull_up(altitude, vSpeed);
-            this.gpws_sink_rate(altitude, vSpeed);
-            this.gpws_dont_sink(altitude, vSpeed, phase);
-            this.gpws_too_low(altitude, Airspeed, FLAP_PushButton, phase);
+        if (radioAlt >= 10 && radioAlt <= 2450 && !SimVar.GetSimVarValue("L:A32NX_GPWS_SYS_OFF", "Bool")) { //Activate between 10 - 2450 radio alt unless SYS is off
+            const FlapPushButton = SimVar.GetSimVarValue("L:A32NX_GPWS_FLAPS3", "Bool");
+            const FlapPosition = SimVar.GetSimVarValue("FLAPS HANDLE INDEX", "Number");
+            const FlapsInLandingConfig = FlapPushButton ? (FlapPosition === 3) : (FlapPosition === 4);
+            const vSpeed = Simplane.getVerticalSpeed();
+            const Airspeed = SimVar.GetSimVarValue("AIRSPEED INDICATED", "Knots");
+            const gearExtended = SimVar.GetSimVarValue("GEAR TOTAL PCT EXTENDED", "Percent") > 0.9;
+
+            this.update_maxRA(radioAlt);
+
+            this.GPWSMode1(radioAlt, vSpeed);
+            //this.GPWSMode2(radioAlt, Airspeed, FlapsInLandingConfig, gearExtended);
```

Comment:
We should add a comment saying why this is commented out.

Reply:
True. Will add it.